### DISCUSSION TITLE
How to use ansible-semaphore with terraform

### DISCUSSION THREAD

**Discussion by bioinfornatics** [semaphoreui/semaphore#889] [Q&A] [answered]
Dear,

Thanks for your amazing tools you have done a nice works.

After read documentation and test the demo it is unclear to me how to use terraform and ansible together through semaphore.
Indeed most of time the inventory will be generated by terraform output which will be use by the ansible playbook to set vars and inventory.

Thanks for your highlights

Have a nice day

**Comment by fiftin** [comment]
Hi @bioinfornatics,

Semaphore works only with Ansible. Please provide more details about your use case.

**Comment by jdhayes** [comment, marked as answer]
@bioinfornatics you can create an inventory script that Semephore can parse to gather a list of hosts.
I use Foreman to house my inventory and I have a Python inventory script to gather the hosts from Forman and present them to Semaphore, which is stored in my Ansible playbook as `inventories/production/hosts`.

The general Ansible documentation is here:
https://docs.ansible.com/ansible/latest/user_guide/intro_dynamic_inventory.html

Here is a more specific example for Terraform on GCP:
https://www.percona.com/blog/2021/06/01/how-to-generate-an-ansible-inventory-with-terraform/

Hope that helps.

**Comment by fiftin** [comment]
@bioinfornatics Terraform available in [Cloud](https://cloud.semui.co) and [Premium](https://cloud.semui.co/auth/login?new_project=premium) version of Semaphore.
